docs/platform/nodebalancer/force-all-connections-on-ssl-with-nodebalancers.md

```diff
@@ -0,0 +1,380 @@
+
+        chmod 400 /etc/ssl/localcerts/www.mydomain.com.crt
+
+
+    {: .note }
+    > 
+    > It is an excellent choice to save all of your `.crt` and `.key` files in a secure, offsite location; optionally in a password protected archive file. By doing so, you can recover them later if neccessary.
+
+## Installing the SSL Certificate and Private Key on your NodeBalancer
+
+### Create a configuration profile for Port 443 on your Nodebalancer
+
+
+### Certificate and Private Key
+
+1.  If you select the HTTPS protocol, the **Certificate** and **Private Key** fields will appear.
+
+    [![The NodeBalancer Certificate and Private Key fields.](/docs/assets/1354-nodebalancer_cert.png)](/docs/assets/1354-nodebalancer_cert.png)
+
+2.  Copy the contents of your SSL certificate into the **Certificate** field. If you have concatenated multiple segments of a chained certificate, be sure to copy all of it's contents into the text field, appearing one after the other.
+
+3.  Copy your passphraseless private key into the **Private Key** field.
+
+4.  On your NodeBalancer `Configurations` page, select `Create Configuration`, you will need to create one for each port/protocol that you would like to use, i.e. `80` and `443`.
+
+5.  Under `Edit Configuration`  Once selected, fill out the values in the fields as shown below:
+ 
+         **Port**                    443
+         **Protocol**                HTTPS
+         **Algorithm**               Round Robin
+         **Session Stickiness**      None
+         **Certificate**             < Insert your signed SSL Certificate >
+         **Private Key**             < Insert your Private Key >
+         **Health Check Type**       HTTP Valid Status
+         **Check Interval**          5
+         **Check Timeout**           3
+         **Check Attempts**          2
+         **Check HTTP Path**         /
+
+    Then, select **`Save Changes`**. 
+
+6.  Add as many nodes as you require for the port configuration by selecting **`Add Node`**. Once selected, fill out the values in the fields like so:
+
+         **Label**                   < Backend Linode 1 >
+         **Address**                 < xxx.xxx.xxx.xxx:80 >
+         **Weight**                  100
+         **Mode**                    Accept
+
+    {: .note }
+    > 
+    > In the section labelled **`Address`**,  you will need to replace `xxx.xxx.xxx.xxx` by specify the private IP address that has been provisioned for that particular Linode, followed by a colon and `80`. This is because SSL is terminating at the NodeBalancer.
+
+    Then, select **`Save Changes`**.
+
+
+## Configuring your Web Server with a 301 Redirect
+
+### Configuring your vhost file for the Apache Webserver.
+
+1.  Enable mod_rewrite so that you can redirect all traffic back to the NodeBalancer over port 443/HTTPS. Enter the following command:
+
+        a2enmod rewrite
+     
+    or, you can load the module manually by appending the following to your Apache configuration file:
```

Comment:
>>  Will the first option always work? 

Yes but I am not 100% sure if a2enmod is only for debian based distros? 

>>  If so, no need to present a second option. If not, will the second option always work? 

The second option should work for any distro every time.

>>   If so, no need to present the first option. If not, why, and when would we choose the second option?

I believe that if using debian, a2enmod is the preferred method. But there is no way of doing it in Redhat based distros. Please see:

    --  http://serverfault.com/questions/56394/how-do-i-enable-apache-modules-from-the-command-line-in-redhat